src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -348,21 +343,11 @@ public function set($id, $service)
 
         if ($this->isFrozen()) {
             // setting a synthetic service on a frozen container is alright
-            if (
-                (!isset($this->definitions[$id]) && !isset($this->obsoleteDefinitions[$id]))
-                    ||
-                (isset($this->definitions[$id]) && !$this->definitions[$id]->isSynthetic())
-                    ||
-                (isset($this->obsoleteDefinitions[$id]) && !$this->obsoleteDefinitions[$id]->isSynthetic())
-            ) {
-                throw new BadMethodCallException(sprintf('Setting service "%s" on a frozen container is not allowed.', $id));
+            if (!isset($this->definitions[$id]) || !$this->definitions[$id]->isSynthetic()) {
```

Comment:
However, the good snippet should be:

```diff
- ($this->isFrozen() && !isset($this->definitions[$id]) || !$this->definitions[$id]->isSynthetic()
+ $this->isFrozen() && (!isset($this->definitions[$id]) || !$this->definitions[$id]->isSynthetic())
```

So I guess it was written like this before in order to ease readability.